homework/index.js

```diff
@@ -30,17 +32,100 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderRepository(leftContainer, repository) {
+    leftContainer.innerHTML = '';
+    const table = createAndAppend('table', leftContainer);
+    const tBody = createAndAppend('tBody', table);
+
+    const rows = [
+      {
+        tr: 'table-info',
+        td1: 'Repository: ',
+        td2: [repository.name, repository.html_url],
+      },
+      {
+        tr: 'table-info',
+        td1: 'Description: ',
+        td2: repository.description,
+      },
+      {
+        tr: 'table-info',
+        td1: 'Forks: ',
+        td2: repository.forks,
+      },
+      {
+        tr: 'table-info',
+        td1: 'Updated: ',
+        td2: new Date(repository.updated_at).toLocaleString(),
+      },
+    ];
+
+    rows.forEach(row => {
+      const tr = createAndAppend('tr', tBody, { class: row.tr });
+      createAndAppend('td', tr, { text: row.td1 });
+      if (Array.isArray(row.td2)) {
+        const tdLink = createAndAppend('td', tr);
+        createAndAppend('a', tdLink, { text: row.td2[0], href: row.td2[1] });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        createAndAppend('td', tr, { text: row.td2 });
       }
     });
   }
 
+  function renderContributors(rightContainer, repository, root) {
+    rightContainer.innerHTML = '';
+    const contributorsURL = repository.contributors_url;
+    createAndAppend('p', rightContainer, { text: 'Contributions', id: 'contTitle' });
+    const ul = createAndAppend('ul', rightContainer, { class: 'ul' });
+    fetchJSON(contributorsURL)
+      .catch(err => {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      })
+
+      .then(contributors => {
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, { class: 'li' });
+          const div = createAndAppend('div', li, { class: 'div' });
+          createAndAppend('img', div, { src: contributor.avatar_url, class: 'contImg' });
+          createAndAppend('a', div, {
+            text: contributor.login,
+            href: contributor.html_url,
+            target: '_blank',
+            class: 'contLink',
+          });
+          createAndAppend('p', div, { text: contributor.contributions, class: 'contNumber' });
+        });
+      });
+  }
+
+  function main(url) {
+    const root = document.getElementById('root');
+    const headBox = createAndAppend('div', root, { class: 'header' });
+    createAndAppend('p', headBox, { text: 'HYF Repositories', id: 'headTitle' });
+    const selectBox = createAndAppend('select', headBox, { class: 'selectBox' });
+
+    fetchJSON(url)
+      .catch(err => {
```

Comment:
The `.catch()` method should normally be at the end of the chain. If you place at the beginning of the chain the error is caught, however after that, execution continues with the next `.then()` which should nor run in case of an error. Consequently, if I force an error by making the url invalid I can see this error message in the browser console:

```
Uncaught (in promise) TypeError: Cannot read property 'forEach' of undefined
    at fetchJSON.catch.then.repositories (index.js:112)
```